src/hooks/usePasswordRecovery.tsx

```diff
@@ -0,0 +1,44 @@
+import {ChangeEvent, FormEvent, useState} from "react";
+import {UserEmail} from "../types/UserLoginProps";
+import usersMock from "../mocks/users.mock";
+import Swal from 'sweetalert2'
+
+const usePasswordRecovery = <T extends UserEmail>(initialState: T) => {
+    const [email, setEmail] = useState<T>(initialState);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const isFormValid = (email: UserEmail) => {
+        return usersMock.some((user) => user.email === email);
+    }
+    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setEmail(email => e.target.value as T);
+        setHasError(false);
+    }
+    const onSubmit = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setHasError(!isFormValid(email));
+        if (isFormValid(email)) Swal.fire({
```

Comment:
It looks like the "Ir a Iniciar Sesión" button is not working, maybe related to sweetalert configuration